acceptance/tests/provider/package/yum.rb

```diff
@@ -0,0 +1,96 @@
+
+  agents.each do |agent|
+    yum_command = 'yum'
+
+    step "Setup: Get two available package versions" do
+      on(agent, "yum list -q #{package} --showduplicates | sed -e '/Available Packages/ d' | awk '{print $2}'") do |result|
+        sorted_available_versions = result.stdout.split("\n").sort!.reverse
+        if sorted_available_versions.count >=2
+          higher_package_version = sorted_available_versions[0]
+          lower_package_version = sorted_available_versions[1]
+        end
+      end
+    end
+
+    skip_test "we need at least two package versions to perform the test" unless sorted_available_versions.length > 1
+
+    step "Install a yum package" do
+      package_manifest = resource_manifest('package', package, { ensure: 'present', provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_match(/\s#{package}/, list)
+      end
+      on(agent, "#{yum_command} remove #{package} -y")
+    end
+
+    step "Uninstall a yum package" do
+      on(agent, "#{yum_command} install #{package} -y")
+      package_manifest = resource_manifest('package', package, { ensure: 'absent', provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_no_match(/\s#{package}/, list)
+      end
+    end
+
+    step "Downgrading a yum package" do
+      package_manifest = resource_manifest('package', package, { ensure: higher_package_version, provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_match(/#{higher_package_version}/, list)
+      end
+      package_manifest = resource_manifest('package', package, { ensure: lower_package_version, provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_match(/#{lower_package_version}/, list)
+      end
+      on(agent, "#{yum_command} remove #{package} -y")
+    end
+
+    step "Using semantic versioning to downgrade to a desired version < X" do
+      on(agent, "#{yum_command} install #{package} -y")
+      package_manifest = resource_manifest('package', package, { ensure: "<=#{lower_package_version}", provider: 'yum' } )
+      apply_manifest_on(agent, package_manifest, :catch_failures => true) do
+        list = on(agent, "#{yum_command} list installed").stdout
+        assert_match(/#{lower_package_version}/, list)
+      end
+      # idempotency test
+      package_manifest = resource_manifest('package', package, { ensure: "<=#{lower_package_version}", provider: 'yum' } )
+      exit_code = apply_manifest_on(agent, package_manifest, :catch_changes => true, :acceptable_exit_codes => [ 0 , 2 ]).exit_code
```

Comment:
From my understanding, `:catch_changes` should automatically raise if the exit code is not 0: https://github.com/puppetlabs/beaker-puppet/blob/master/lib/beaker-puppet/helpers/puppet_helpers.rb#L489

So you shouldn't need to get and assert the exit code